docs/user-guide/ide-setup.md

```diff
@@ -0,0 +1,80 @@
+---
+description: Learn how to set up Lite XL as an integrated development environment.
+---
+
+# Setup Lite XL as an IDE
+
+Lite XL can be configured to be a proper IDE for most programming languages.
+The following is a brief overview of the plugins that, when used together, create the experience of a minimal and modern IDE.
+
+## Bells and whistles
+The following plugins are meant to enrich Lite XL with handy features that make coding quicker and more pleasant to the eyes:
+
+- `autoinsert` and `lfautoinsert` add automatic bracket/quote insertion and text-wrapping.
+- `autosave` allows automatic file saving after every change.
+- `bracketmatch` underlines brackets matching pairs under the caret.
+- `colorpreview` underlays color values (eg. `#ff00ff` or `rgb(255, 0, 255)`) with their respective color.
+- `indentguide` adds indent guides to make it easier to see function boundaries.
+- `rainbowparen` applies rainbow colours to nesting parentheses to make it easier to distinguish them.
+- `scalestauts` displays current scale (zoom) in the status view (it depends on the `scale` plugin).
+- `selectionhighlight` highlights regions of code that match the current selection, to preview for search/replace actions.
+- `sticky_scroll` keeps track of the current scope at the top of the view.
+- `minimap` adds VSCode's minimap.
+- `devicons` provides fancy icons in the project treeview for many different file types.
+- `meta_colors` installs all available theme files
+
+## Syntax highlighting
+`meta_languages` is a metapackage that installs all supported syntax-highlighting plugins.
+
+## Intellisense
+`lsp` is a plugin that provides support for intellisense (LSP protocol), while `lintplus` provides support for linters and is used when the an lsp server lacks an integrated linter.
```

Comment:
```suggestion
`lsp` is a plugin that provides support for intellisense (LSP protocol), while `lintplus` provides support for linters in case A LSP server lacks an integrated linter.
```